Backfill Runtime release notes (4.0.9 - 4.0.11) (#414)
astronomer/docs: astro/runtime-release-notes.md

For instructions on how to upgrade, read [Upgrade Astro Runtime](upgrade-runtime.md). For general product release notes, go to [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, reach out to [Astronomer Support](https://support.astronomer.io).

## Astro Runtime 4.0.11

- Release date: February 14, 2022
- Airflow version: 2.2.3

> ### Additional Improvements
> 
> - Bumped `openlineage-airflow` to `0.5.2`

Reviewer: ```suggestion
### Additional Improvements

- Upgraded the `openlineage-airflow` library in Astro Runtime to `v0.5.2`
```

Reviewer: In the future I'd love to spell this out to include what's encompassed in the version bump here, but this is fine for now

Author: Would love to if we had any time! Updating this to not include reference to "Astro Runtime", as that's given

## Astro Runtime 4.0.10

- Release date: February 9, 2022
- Airflow version: 2.2.3

### Additional Improvements

- The Airflow UI now shows a project's Runtime version in the footer of all pages:

>     ![Runtime Version banner](runtime-version.png)

Reviewer 2: This image does not contain the Image tag that @paolaperaza suggested on the lines below. Example

![image](https://user-images.githubusercontent.com/8811558/155532312-f8ee8c56-797a-4a10-98ee-011ec60a2e94.png)

Reviewer 2: (Image Source: https://github.com/astronomer/airflow-operator/issues/606#issuecomment-1041593917)

Author: Ah I see now, I managed to get this working in dev but not on production. Might mean that there's another piece here that needs to be fixed in Astro before it works in Runtime?
<img width="523" alt="Screen Shot 2022-02-24 at 1 48 26 PM" src="https://user-images.githubusercontent.com/74574233/155536692-06bb0a22-769f-4fbb-91df-6e9405b4f917.png">

Reviewer 2: Yeah, the new version of Operator might need to be rolled out to Prod.

cc @ianstanton who might know

Reviewer 3: Infra team had to delay the internal prod data-plane rollout containing this feature until next week. It should be present in dev and stage.

Reviewer: Ah, thanks @ianstanton - will touch base with infra team and remove from release notes this week

> ### Additional Improvements
> 
> - The Airflow UI now shows a project's Runtime version in the footer of all pages:
> 
>     ![Runtime Version banner](runtime-version.png)
> 
> - Updated all Astronomer operators to version `1.1.0`.
> - The `apache-airflow-providers-elasticsearch` provider package is now installed by default.

Reviewer: ```suggestion
### Running Docker Image Tag in Airflow UI

The Airflow UI now shows the Docker image tag that is running on the Airflow Webserver at any given time. Depending on how your team deploys to Astro, this value is either a unique identifier generated by a CI tool or a timestamp generated by the Astro CLI on `$ astrocloud deploy`. Both represent a unique version of your Astro project.

    ![Runtime Version banner](runtime-version.png)

When you push code to a Deployment on Astro via the Astro CLI or CI/CD, reference this tag in the Airflow UI to verify that your changes were successfully applied.

::: info

The Docker image tag shown in the Airflow UI only represents the tag that is running on at least 1 Webserver container in your Deployment. While it is a likely indicator, it is not a guarantee that this Docker image and version of your Astro project is running on your Airflow Scheduler, Workers, or Triggerer.

If you ever have trouble verifying a code push to a Deployment on Astro, reach out to [Astronomer Support](https://support.astronomer.io). We're here to help.

:::

### New Deferrable Operators Now Available

Astro Runtime now includes the following operators:

- KubernetesPodOperatorAsync
- HttpSensorAsync
- SnowflakeOperatorAsync
- FileSensorAsync

These are all [deferrable operators](deferrable-operators.md) built by Astronomer and available exclusively on Astro Runtime. They are pre-installed into the Astro Runtime Docker image and are ready to use.

### Additional Improvements

- To support an enhanced logging experience on Astro, the `apache-airflow-providers-elasticsearch` provider package is now installed by default.
```

Reviewer: @jwitz SO.... I don't actually see this in the image you shared, but it looks like you're running the right version and pushed code to it. Can you work with Ian Stanton if you haven't already to verify that everything we need for this feature is in there?

Reviewer 2: `Astro Runtime now includes the following operators:` -> `Astro Runtime now also includes the following operators:`. Just adding "also" to it, since we are only listing the new operators and excluding the ones that were already present like `DatabricksSubmitRunOperatorAsync` and `ExternalTaskSensorAsync`

Reviewer: Good call, agreed - made a note here

## Astro Runtime 4.0.9

- Release date: January 19, 2022
- Airflow version: 2.2.3

> ### Additional Improvements
> 
> - The `openlineage-airflow` provider package is now installed in Runtime by default.

Reviewer: ```suggestion
### New OpenLineage and Airflow Library

Astronomer is committed to providing a strong data lineage experience on Astro. As a first step, the `openlineage-airflow` provider package is now pre-installed in Astro Runtime by default. This package allows users to automatically send OpenLineage events to a data lineage backend, like [Marquez](https://marquezproject.github.io/marquez/), if configured.

Over the next few months, our team will be building out an even stronger native integration with data lineage tooling. For more information, read [OpenLineage documentation](https://openlineage.io/integration/apache-airflow/) or learn from our [Data Lineage with Airflow webinar](https://www.astronomer.io/events/recaps/data-lineage-with-openlineage-and-airflow/).
```

Reviewer: Runtime release are rarely going to have huge features, so let's try to call out these high-level value adds more explicitly and not have "Additional Improvements" be the only thing that's on there

Author: I don't think the solution to not having huge features is having huge release notes. Look at [Cloud Composer release notes](https://cloud.google.com/composer/docs/release-notes) and [Databricks release notes](https://docs.databricks.com/release-notes/runtime/9.0.html#optimized-image-data-storage). All of the content is directly related to the change/ feature and gets to the core of what users need to know about

We really need a platform to communicate these changes outside of technical documentation. I had no idea that we were even ready to start referring to our lineage integration in documentation. It's neither sustainable nor useful to have release notes be a crutch for lack of communication/ features elsewhere.

Reviewer: Talked this out over Zoom, but I think we agreed there's a middle ground here. Let's cut back on the open lineage statement (recent changes in this PR look good), but _while_ we don't have a developer newsletter or a place to announce product releases to our customers, release notes are the only way to communicate the value of the changes we're making.

We'll get there 🚀

For instructions on how to upgrade, read [Upgrade Astro Runtime](upgrade-runtime.md). For general product release notes, go to [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, reach out to [Astronomer Support](https://support.astronomer.io).

## Astro Runtime 4.0.11

- Release date: February 14, 2022
- Airflow version: 2.2.3

### Additional Improvements

- Upgraded the `openlineage-airflow` library to `v0.5.2`

## Astro Runtime 4.0.10

- Release date: February 9, 2022
- Airflow version: 2.2.3

### Running Docker Image Tag in Airflow UI

The Airflow UI now shows the Docker image tag that is running on the Airflow Webserver at any given time. Depending on how your team deploys to Astro, this value is either a unique identifier generated by a CI tool or a timestamp generated by the Astro CLI on `$ astrocloud deploy`. Both represent a unique version of your Astro project.

![Runtime Version banner](runtime-version.png)

When you push code to a Deployment on Astro via the Astro CLI or CI/CD, reference this tag in the Airflow UI to verify that your changes were successfully applied.

::: info

> The Docker image tag shown in the Airflow UI only represents the tag that is running on at least 1 Webserver container in your Deployment. While it is a likely indicator, it is not a guarantee that this Docker image and version of your Astro project is running on your Airflow Scheduler, Workers, or Triggerer.
> If you ever have trouble verifying a code push to a Deployment on Astro, reach out to [Astronomer Support](https://support.astronomer.io).

Reviewer: ```suggestion
The Docker image tag shown in the Airflow UI only represents the tag that is running on at least 1 Webserver container in your Deployment. While it is a likely indicator, it is not a guarantee that this Docker image and version of your Astro project is running on your Airflow Scheduler, Workers, or Triggerer.

If you ever have trouble verifying a code push to a Deployment on Astro, reach out to [Astronomer Support](https://support.astronomer.io).
```

:::

### New Deferrable Operators Now Available

> Astro Runtime now includes the following operators:

Reviewer: ```suggestion
Astro Runtime now also includes the following operators:
```

- `KubernetesPodOperatorAsync`
- `HttpSensorAsync`
- `SnowflakeOperatorAsync`
- `FileSensorAsync`

> These are all [deferrable operators](deferrable-operators.md) built by Astronomer and available exclusively on Astro Runtime. They are pre-installed into the Astro Runtime Docker image and are ready to use.

Reviewer: ```suggestion
These are all [deferrable operators](deferrable-operators.md) built by Astronomer and available exclusively on Astro Runtime. They are pre-installed into the Astro Runtime Docker image and ready to use.
```
